### PR TITLE
[denial-of-merge attack] Should prevent food from runtiming because of null reagents

### PR DESCRIPTION
**Pull request by Wizardcrying**
Gives a few sanity checks to make sure that no insane shit's going on
Also removes a spawn
This doesn't seem like very good code but I'm tired so
Goodnight

### CODE REVIEW

**Comment by clusterfack** [comment]
Sanity isn't really a fix for this sort of thing, it's better to have the runtime that indicates there is a problem

File: code/modules/reagents/reagent_containers/food/snacks.dm [hunk 1]
```diff
@@ -159,9 +163,7 @@
 						reagents.trans_to(target, reagents.total_volume)
 					bitecount++
 					On_Consume(target)
-			return 1
```
Comment (clusterfack) [inline code/modules/reagents/reagent_containers/food/snacks.dm]: Why remove this

Comment (Wizardcrying) [inline code/modules/reagents/reagent_containers/food/snacks.dm:162]: What would returning 1 do?

**Comment by Wizardcrying** [comment]
@clusterfack But what should I fix, then? Because the snacks are still quite obviously trying to transfer null reagents.

File: code/modules/reagents/reagent_containers/food/snacks.dm (hunk 1 above)
Comment (clusterfack) [inline code/modules/reagents/reagent_containers/food/snacks.dm]: If you don't know why did you remove it, it prevents the next proc called in the chain of attack thus it should block afterattack from being called.

**Comment by clusterfack** [comment]
Look for the stack trace in the runtime and see what is calling it in this way

**Comment by clusterfack** [comment]
I've fixed this in a separate PR
